compiler/rustc_macros/src/diagnostics/diagnostic_builder.rs

```diff
@@ -193,59 +205,116 @@ impl<'a> DiagnosticDeriveVariantBuilder<'a> {
         let name = attr.path().segments.last().unwrap().ident.to_string();
         let name = name.as_str();
 
-        let mut first = true;
+        let mut set_label = false;
+        let keys = vec!["note", "help", "warning", "suggestion"];
 
         if name == "diag" {
             let mut tokens = TokenStream::new();
-            attr.parse_nested_meta(|nested| {
-                let path = &nested.path;
-
-                if first && (nested.input.is_empty() || nested.input.peek(Token![,])) {
-                    self.slug.set_once(path.clone(), path.span().unwrap());
-                    first = false;
-                    return Ok(());
+            match &attr.meta {
+                // support syntax `#[diag("message ...", code = "E0045", note = "node message")]`
+                Meta::List(MetaList { path, tokens: token_stream, .. }) => {
```

Comment:
This is to support:
```rust
#[diag("message....")] // or #[diag("message", code = "...", note = "...")]
```

If it's:
```rust
#[diag(label = "message....")] 
```
it will be handled by `attr.parse_nested_meta` below.

I'd suggest we only support the late one, which is more consistent with `suggestion`, `multipart_suggestion`, or we only support `#[diag("message....")]`(without extra nested elements), if there are multiple nested elements `label = ` is required?